app/src/org/commcare/views/widgets/CommCareAudioWidget.java

```diff
@@ -105,88 +114,182 @@ public IAnswerData getAnswer() {
+            int mCurrentPosition = player.getCurrentPosition();
+            playbackSeekBar.setProgress(mCurrentPosition / 1000);
+            playbackTime.setText(getTimeString(mCurrentPosition));
+        }
+    }
+
+
+    private String getTimeString(long millis) {
+        StringBuffer buf = new StringBuffer();
+
+        int hours = (int)(millis / (1000 * 60 * 60));
+        int minutes = (int)((millis % (1000 * 60 * 60)) / (1000 * 60));
+        int seconds = (int)(((millis % (1000 * 60 * 60)) % (1000 * 60)) / 1000);
+
+        if (hours > 0) {
+            buf.append(String.format(Locale.getDefault(), "%02d", hours)).append(":");
+        }
+
+        buf.append(String.format(Locale.getDefault(), "%02d", minutes)).append(":")
+                .append(String.format(Locale.getDefault(), "%02d", seconds));
+
+        return buf.toString();
     }
 
     private void pauseAudioPlayer() {
         player.pause();
         mPlayButton.setBackgroundResource(R.drawable.play);
         mPlayButton.setOnClickListener(v -> resumeAudioPlayer());
+        stopPlaybackTimer();
     }
 
     private void resumeAudioPlayer() {
         player.start();
         mPlayButton.setBackgroundResource(R.drawable.pause);
         mPlayButton.setOnClickListener(v -> pauseAudioPlayer());
+        startPlaybackTimer();
     }
 
     private void resetAudioPlayer() {
-        player.release();
-        mPlayButton.setBackgroundResource(R.drawable.play);
-        mPlayButton.setOnClickListener(v -> playAudio());
+        if (player != null) {
+            player.release();
+        }
+        stopPlaybackTimer();
     }
 
     @Override
     protected void togglePlayButton(boolean enabled) {
         if (enabled) {
-            mPlayButton.setBackgroundResource(R.drawable.play);
+            initAudioPlayer();
+            captureButton.setBackgroundResource(R.drawable.recording_trash);
         } else {
-            mPlayButton.setBackgroundResource(R.drawable.play_disabled);
+            resetAudioPlayer();
+            hidePlaybackIndicators();
+            captureButton.setBackgroundResource(R.drawable.record);
         }
-        mPlayButton.setEnabled(enabled);
+    }
+
+    private void hidePlaybackIndicators() {
+        mPlayButton.setVisibility(INVISIBLE);
+        playbackSeekBar.setVisibility(INVISIBLE);
+        playbackDuration.setVisibility(INVISIBLE);
+        playbackTime.setVisibility(INVISIBLE);
+    }
+
+    private void initAudioPlayer() {
+        mPlayButton.setVisibility(VISIBLE);
+        mPlayButton.setBackgroundResource(R.drawable.play);
+        mPlayButton.setOnClickListener(v -> playAudio());
+
+        Uri filePath = Uri.parse(mInstanceFolder + mBinaryName);
+        player = MediaPlayer.create(getContext(), filePath);
+        player.setOnCompletionListener(mp -> onCompletePlayback());
+
+        playbackDuration.setVisibility(VISIBLE);
+        playbackDuration.setText("/" + getTimeString(player.getDuration()));
```

Comment:
Very minor feedback: If this was a String.format() input, it would be easier to update / localize rather than a concatenated string